packages/rest/src/parser.ts

```diff
@@ -83,7 +83,7 @@ async function buildOperationArguments(
     }
     const spec = paramSpec as ParameterObject;
     const rawValue = getParamFromRequest(spec, request, pathParams);
-    const coercedValue = coerceParameter(rawValue, spec);
+    const coercedValue = await coerceParameter(rawValue, spec);
```

Comment:
Building on top of my previous comments, perhaps this is the place where to call validation?

```suggestion
    const coercedValue = coerceParameter(rawValue, spec);
    const finalValue = await validateParameterValue(rawValue, spec);
    paramArgs.push(finalValue);
```

(Of course, this will not help with collecting all validation error together and other aspects mentioned in my other comments, but it may be good enough as a short-term fix.)

Reply:
We have also access to `globalSchemas` and `options` here, which is needed to resolve `{$ref}` schemas and use the same AJV factory as the request body validator does.

```suggestion
    const coercedValue = await coerceParameter(rawValue, spec);
    const coercedValue = coerceParameter(rawValue, spec);
    const finalValue = await validateParameterValue(rawValue, spec, globalSchemas, options);
    paramArgs.push(finalValue);
```

Reply:
Thanks @bajtos I tried calling `validateValueAgainstSchema` in the new place, and got 20+ test failures. This story is to coerce the **object** based on schema, but `coerceParameter` handles all types, and same to `validateValueAgainstSchema` if I call it after `coerceParameter`. 

I am a bit reluctant to investigate and verify why other types fail, and considering `schema` is usually used for object, can we deal with other types next time?